power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -36,16 +38,127 @@ static_assert(is_indptr_mutable_v<writable_dataset_t>);
 struct ComponentInfo {
     MetaComponent const* component;
     // for non-uniform component, this is -1, we use indptr to describe the elements per scenario
-    Idx elements_per_scenario;
-    Idx total_elements;
+    Idx elements_per_scenario{};
+    Idx total_elements{};
 };
 
 struct DatasetInfo {
-    bool is_batch;
-    Idx batch_size; // for single dataset, the batch size is one
-    MetaDataset const* dataset;
+    bool is_batch{false};
+    Idx batch_size{1}; // for single dataset, the batch size is one
+    MetaDataset const* dataset{nullptr};
     std::vector<ComponentInfo> component_info;
 };
+template <typename Data> struct AttributeBuffer {
```

Comment:
use `class` instead of `typename` to be consistant with the rest of the code. Please do so in all the places.

Reply:
Done.